tests/test_client.py

```diff
@@ -536,3 +536,12 @@ async def test_client_connecting_disconnected_done() -> None:
     client._disconnected.set_result(None)
     await client.connect()
     await client.disconnect()
+
+
+@pytest.mark.network
+async def test_client_aenter_connect_error_lock_release() -> None:
+    client = Client(hostname="aenter_connect_error_lock_release")
+    with pytest.raises(MqttError):  # noqa: PT012
+        async with client:
```

Comment:
You could use `await client.__aenter__()` here instead, this way we do not have to ignore the rule. Otherwise, good test! 👍